server/src/graql/answer/ConceptMap.java

```diff
@@ -18,129 +18,311 @@
+        return map;
+    }
+
+    @javax.annotation.CheckReturnValue
+    public Set<Var> vars() { return map.keySet();}
+
+    @javax.annotation.CheckReturnValue
+    public Collection<Concept> concepts() { return map.values(); }
 
     /**
      * Return the {@link Concept} bound to the given variable name.
      *
      * @throws GraqlQueryException if the {@link Var} is not in this {@link ConceptMap}
      */
-    @CheckReturnValue
-    Concept get(String var);
+    @javax.annotation.CheckReturnValue
+    public Concept get(String var) {
+        return get(Graql.var(var));
+    }
 
     /**
      * Return the {@link Concept} bound to the given {@link Var}.
      *
      * @throws GraqlQueryException if the {@link Var} is not in this {@link ConceptMap}
      */
-    @CheckReturnValue
-    Concept get(Var var);
+    @javax.annotation.CheckReturnValue
+    public Concept get(Var var) {
+        Concept concept = map.get(var);
+        if (concept == null) throw GraqlQueryException.varNotInQuery(var);
+        return concept;
+    }
 
-    @CheckReturnValue
-    boolean containsVar(Var var);
+    @javax.annotation.CheckReturnValue
+    public boolean containsVar(Var var) { return map.containsKey(var);}
 
-    @CheckReturnValue
-    boolean containsAll(ConceptMap ans);
+    @javax.annotation.CheckReturnValue
+    public boolean containsAll(ConceptMap map) { return this.map.entrySet().containsAll(map.map().entrySet());}
 
-    @CheckReturnValue
-    boolean isEmpty();
+    @javax.annotation.CheckReturnValue
+    public boolean isEmpty() { return map.isEmpty();}
 
-    @CheckReturnValue
-    int size();
+    @javax.annotation.CheckReturnValue
+    public int size() { return map.size();}
 
-    void forEach(BiConsumer<Var, Concept> consumer);
+    @Override
+    public String toString() {
+        return map.entrySet().stream()
+                .sorted(Comparator.comparing(e -> e.getKey().getValue()))
+                .map(e -> "[" + e.getKey() + "/" + e.getValue().id() + "]").collect(Collectors.joining());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+        ConceptMap a2 = (ConceptMap) obj;
+        return map.equals(a2.map);
+    }
+
+    @Override
+    public int hashCode() { return map.hashCode();}
+
+    public void forEach(BiConsumer<Var, Concept> consumer) {
+        map.forEach(consumer);
+    }
 
     /**
-     * perform an answer merge without explanation
+     * perform an answer merge with optional explanation
      * NB:assumes answers are compatible (concept corresponding to join vars if any are the same)
      *
-     * @param a2 answer to be merged with
+     * @param a2          answer to be merged with
+     * @param explanation flag for providing explanation
      * @return merged answer
      */
-    @CheckReturnValue
-    ConceptMap merge(ConceptMap a2);
+    @javax.annotation.CheckReturnValue
```

Comment:
we're refering to the annotation directly rather than importing like this:

```
import javax.annotation.CheckReturnValue;
...

@CheckReturnValue
```